dali/image/tiff.cc

```diff
@@ -0,0 +1,69 @@
+#include "tiff.h"
+
+namespace dali {
+
+namespace {
+
+cv::Mat DecodeTiff(const unsigned char *tiff, int size, DALIImageType image_type) {
+    std::vector<char> buf(tiff, tiff + size);
+    return cv::imdecode(buf, image_type == DALI_GRAY ? 0 : 1);
+}
+
+
+} // namespace
+
+bool CheckIsTiff(const unsigned char *tiff) {
+    assert(tiff);
+
+    std::vector<int> header_intel = {77, 77, 0, 42};
+    std::vector<int> header_motorola = {73, 73, 42, 0};
+
+    auto check_header = [&](const std::vector<int> &header) -> bool {
+        for (unsigned int i = 0; i < header.size(); i++) {
+            if (tiff[i] != header[i]) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    return check_header(header_intel) || check_header(header_motorola);
+}
+
+
+DALIError_t GetTiffImageDims(const unsigned char *tiff, int size, int *h, int *w) {
+    assert(CheckIsTiff(tiff));
+    auto tiff_mat = DecodeTiff(tiff, size, DALI_RGB);
+    if (tiff_mat.empty()) {
+        return DALIError;
+    }
+    *h = tiff_mat.rows;
+    *w = tiff_mat.cols;
+    return DALISuccess;
+}
+
+
+DALIError_t DecodeTiffHost(const unsigned char *tiff, int size, DALIImageType image_type, Tensor<CPUBackend> *output) {
```

Comment:
This code path, similarly to DecodePngHost is the same (or I'm missing the difference) as fallback to OpenCV in the HostDecoder.

Reply:
Same thing as above, I'll do some refactoring here